mlos_bench/mlos_bench/services/remote/azure/azure_auth.py

```diff
@@ -24,7 +24,7 @@ class AzureAuthService(Service, SupportsAuth):
 
     _REQ_INTERVAL = 300   # = 5 min
 
-    def __init__(self, config: dict, parent: Service):
+    def __init__(self, config: dict, global_config: dict, parent: Service):
```

Comment:
Not the end of the world probably, but we switch at some point from these being optional and declared with types to just `dict`.  Same for `parent`.

Reply:
Would be good to be consistent or document why not.

Reply:
You are right. This PR is a good opportunity to make all constructors consistent - at least for the Service class hierarchy. I'll push an update soon and let you know.